bitswap.go

```diff
@@ -229,7 +225,7 @@ func (bs *Bitswap) GetBlocks(ctx context.Context, keys []cid.Cid) (<-chan blocks
 		log.Event(ctx, "Bitswap.GetBlockRequest.Start", k)
 	}
 
-	mses := bs.getNextSessionID()
+	mses := bs.sm.GetNextSessionID()
```

Comment:
One thing to note: We should be moving towards having *only* sessions. These two methods on the main bitswap interface are really just legacy shims that we should be moving away from. The weirdness of the abstractions right now is a symptom of that (Why are we leaking session IDs out of the session manager?)

Reply:
that makes sense. also though, to be clear, the sessions implemented in sessions.go seem to be different than the session used here -- I'm not even entirely clear what it does in this case? (since it doesn't seem to limit peers in the same way a full session does). Would love some clarity on this part of the code cause I don't believe I fully understand it yet.

Reply:
The reason that we're using a session ID here is because wants in the wantmanager get tracked by session. Since multiple different sessions may want the same hash, we don't want one session cancelling their request to remove the want from the wantlist for a different session.

Reply:
Got it, but just to check for understanding, there is in fact no active real "session" per se -- just want to make sure I understand. So this is almost analogous to a "requestor" - where requestor could be a direct call to Bitswap's GetBlocks w/o a session, or it could be a real longer lived session that might make many requests over time.

Reply:
But is the plan to have bitswap's GetBlocks actually use a real session as well (if the caller doesn't initialize a session directly and calls GetBlocks w/o a session)? Is there something preventing us from doing that now?